### PR TITLE
PartialOrd between Z, Q and  Modulus

### PR DESCRIPTION
**Pull request by Marcel583**
**Description**

This PR implements .

**Testing**

- [x] I added basic working examples (possibly in doc-comment)
- [x] I added tests for large (pointer representation) values
- [x] I triggered all possible errors in my test in every possible way
- [x] I included tests for all reasonable edge cases

**Checklist:**

- [x] I have performed a self-review of my own code
  - [x] The code provides good readability and maintainability s.t. it fulfills best practices like talking code, modularity, ...
    - [x] The chosen implementation is not more complex than it has to be
  - [x] My code should work as intended and no side effects occur (e.g. memory leaks)
  - [x] The doc comments fit our style guide
  - [x] I have credited related sources if needed

### CODE REVIEW

File: src/rational/q/cmp.rs [hunk 1]
```diff
@@ -599,6 +671,53 @@ mod test_partial_ord {
     }
 }
 
+/// Test that the [`PartialOrd`] trait is correctly implemented.
+#[cfg(test)]
+mod test_partial_ord_q_other {
+    use super::Q;
+    use crate::{integer::Z, integer_mod_q::Modulus};
+
+    // Ensure that the function can be called with several types
```
Comment (jnsiemer) [inline src/rational/q/cmp.rs:680]:
```suggestion
    /// Ensure that the function can be called with several types
```

File: src/macros/for_others.rs [hunk 2]
```diff
@@ -96,6 +99,27 @@ macro_rules! implement_for_others {
...
+        $(#[doc(hidden)] impl PartialOrd<$source_type> for $type {
+            paste::paste! {
+                #[doc = "Documentation can be found at [`" $type "::partialEq`]."]
+                fn partial_cmp(&self, other: &$source_type) -> Option<Ordering> {
+                    self.partial_cmp(&$bridge_type::from(other))
+                }
+            }
+        }
+
+        #[doc(hidden)] impl PartialOrd<$type> for $source_type {
```
Comment (jnsiemer) [inline src/macros/for_others.rs:114]: Can't we take two additional input parameters for the trait and the function s.t. we don't have to write another part to this macro?

File: src/integer_mod_q/modulus/cmp.rs [hunk 3]
```diff
@@ -196,6 +324,46 @@ mod test_partial_ord {
     }
 }
 
+/// Test that the [`PartialOrd`] trait is correctly implemented.
+#[cfg(test)]
+mod test_partial_ord_modulus_other {
+    use super::Modulus;
+    use crate::integer::Z;
+
+    // Ensure that the function can be called with several types
```
Comment (jnsiemer) [inline src/integer_mod_q/modulus/cmp.rs:333]:
```suggestion
    /// Ensure that the function can be called with several types
```

File: src/integer_mod_q/modulus/cmp.rs [hunk 4]
```diff
@@ -159,6 +235,58 @@ mod test_eq {
...
+        assert!(2i64 == modulus);
+        assert!(2u64 == modulus);
+
+        assert!(&modulus == &z);
+        assert!(&z == &modulus);
+        assert!(&modulus == &2i8);
+        assert!(&2i8 == &modulus);
+    }
+
+    // Ensure that large values are compared correctly
```
Comment (jnsiemer) [inline src/integer_mod_q/modulus/cmp.rs:279]:
```suggestion
    /// Ensure that large values are compared correctly
```

File: src/integer_mod_q/modulus/cmp.rs [hunk 5]
```diff
@@ -159,6 +235,58 @@ mod test_eq {
     }
 }
 
+/// Test that the [`PartialEq`] trait is correctly implemented.
+#[cfg(test)]
+mod test_partial_eq_modulus_other {
+    use super::Z;
+    use crate::integer_mod_q::Modulus;
+
+    // Ensure that the function can be called with several types
```
Comment (jnsiemer) [inline src/integer_mod_q/modulus/cmp.rs:244]:
```suggestion
    /// Ensure that the function can be called with several types
```

**Review by jnsiemer — APPROVED** [review]
Looks good!

File: src/macros/for_others.rs (hunk 2 above)
Comment (Marcel583) [reply src/macros/for_others.rs:114]: It is not so simple, because for example the mul trait needs an output type definition and PartialEq and PartialOrd have different output types.
